enhancements/network/user-defined-network-segmentation.md

```diff
@@ -413,6 +519,203 @@ guarantee uniqueness and eliminates the ability to falsify access.
+          "excludeSubnets":"10.0.0.100/24",
+          "mtu":1500,   
+          "name":"8301b3c4-ff07-4c6d-9786-232e02b87794",  <--- generated unique network-name
+          "netAttachDefName":"demo/poc-db-network",
+          "subnets":"10.0.0.0/24",
+          "topology":"layer2",
+          "type":"ovn-k8s-cni-overlay",
+          "primaryNetwork": "true"
+      }'
+```
+
+Creating cluster scoped CRD instance should trigger creation of the corresponding NAD at each namespace specified in the spec.
+Following the above cluster-scope CRD [example](#example---namespace-scoped-network), the following NADs should be created:
+```yaml
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: db-network
+  namespace: mynamespace
+  finalizers:
+  - k8s.ovn.org/userdefinednetwork
+  ownerReferences:
+    - apiVersion: k8s.ovn.org/v1alpha1
+      blockOwnerDeletion: true
+      kind: ClusterUserDefinedNetwork
+      name: db-network
+      uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >    
+      '{
+          "cniVersion":"0.3.1",
+          "excludeSubnets":"10.0.0.100/24",
+          "mtu":1500,   
+          "name":"ef1df435-ef21-4850-a4a2-5fd8a8a8f991",   <--- generated unique network-name
+          "netAttachDefName":"mynamespace/db-network",
+          "subnets":"10.0.0.0/24",
+          "topology":"layer2",
+          "type":"ovn-k8s-cni-overlay",
+          "primaryNetwork": true
+      }'
+---
+apiVersion: k8s.cni.cncf.io/v1
+kind: NetworkAttachmentDefinition
+metadata:
+  name: db-network
+  namespace: theirnamespace
+  finalizers:
+  - k8s.ovn.org/userdefinednetwork
+  ownerReferences:
+  - apiVersion: k8s.ovn.org/v1alpha1
+    blockOwnerDeletion: true
+    kind: ClusterUserDefinedNetwork
+    name: db-network
+    uid: f45efb13-9511-48c1-95d7-44ee17c949f4
+spec:
+  config: >
+    '{
+        "cniVersion":"0.3.1",
+        "excludeSubnets":"10.0.0.100/24",
+        "mtu":1500,   
+        "name":"ef1df435-ef21-4850-a4a2-5fd8a8a8f991",   <--- same name as in other namespaces
+        "netAttachDefName":"theirnamespace/db-network",
+        "subnets":"10.0.0.0/24",
+        "topology":"layer2",
+        "type":"ovn-k8s-cni-overlay",
+        "primaryNetwork": true
+    }'
+```
+
+##### Validations
+The controller should validate the request CRD spec and verify:
+- IP addresses, CIDRs are valid.
+- MTU is valid, no more then 9000.
+- `topology` is one of `Layer2` or `Layer3`.
+- `role` is one of `Private` or `Secondary`.
+
+In addition, the following scenarios should be validated: 
+- When private network is requested (i.e.: `spec.role: Private`)
+  - Verify no private network exist at the target namespace (i.e.: no NAD with `"privateNetwork": "true"` exist).
+    In case private-network already exist, the request CR status should reflect network is not ready because private network
+    already exist at the target namespace.
+
+- When a managed NAD already exist at the target namespace:
```

Comment:
One question; this point will cover both the following scenarios right ? 
- there's a user defined network (namespace scoped) in a namespace. The cluster admin then creates a cluster-wide network that clashes with it. 
- there's a cluster defined network (defined by the cluster admin). The user then creates namespaced scoped netwrk that clashes with it. 

I.e. we do not differentiate between clashes originated via a pre-existing NAD from clashes created via a pre-existing conflicting CRD. I think that's OK, just want to make sure it is.